File: TransCoder_tokenized_test_set_functions/python/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL.py
def FirstRepeated ( string ) :
    checker = 0
    pos = 0
    for i in string :
        val = ord ( i ) - ord ( 'a' );
        
        if ( ( checker & ( 1 << val ) ) > 0 ) :
            return pos
        checker |= ( 1 << val )
        pos += 1
    return - 1

File: TransCoder_tokenized_test_set_functions/java/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL.java
static int FirstRepeated ( String str ) {
  int checker = 0;
  for ( int i = 0;
  i < str . length ( );
  ++ i ) {
    int val = ( str . charAt ( i ) - 'a' );
    if ( ( checker & ( 1 << val ) ) > 0 ) return i;
    checker |= ( 1 << val );
  }
  return - 1;
}